projects/vdk-core/tests/vdk/internal/builtin_plugins/ingestion/test_ingester_base.py

```diff
@@ -184,3 +184,37 @@ def test_ingest_payload_multiple_destinations():
         destination_table=destination_table2,
         payload=test_expected_payload2,
     )
+
+
+def test_ingest_payload_and_post_ingestion_operation():
+    test_payload = {"key1": "val1", "key2": "val2", "key3": "val3"}
+    test_aggregated_payload = [{"key1": "val1", "key2": "val2", "key3": "val3"}]
+    test_ingestion_result = {"some_metadata": "some_ingestion_metadata"}
+    destination_table = "a_destination_table"
+    method = "test_method"
+    target = "some_target"
+    collection_id = "test_job|42a420"
+    ingester_base = create_ingester_base()
+
+    ingester_base._ingester.ingest_payload.return_value = test_ingestion_result
+
+    ingester_base.send_object_for_ingestion(
+        payload=test_payload,
+        destination_table=destination_table,
+        method=method,
+        target=target,
+    )
+    ingester_base.close()
+
+    ingester_base._ingester.ingest_payload.assert_called_once()
+    ingester_base._ingester.ingest_payload.assert_called_with(
+        payload=test_aggregated_payload,
+        destination_table=destination_table,
+        target=target,
+        collection_id=collection_id,
+    )
+    ingester_base._ingester.post_ingest_process.assert_called_with(
+        payload=test_aggregated_payload,
+        ingestion_result=test_ingestion_result,
+        exceptions=[],
```

Comment:
Maybe add a test where an exception is propagated to the post-ingest hook implementation?

Reply:
Added, thanks